### ISSUE TITLE
Cannot click on "Settings" when running in browser

### ISSUE THREAD

**Issue by yaooluu** [tombaker1/EdenMobile#1]
Hi Tom!

I followed your guide to run EdenMobile in browser, but the "Settings" link is unclickable. Below is all the error I got from the browser:
![1](https://cloud.githubusercontent.com/assets/8813134/6772916/d9f0653a-d0de-11e4-9929-c44c7f2f1569.PNG)
(Omitted some log info between...)
![2](https://cloud.githubusercontent.com/assets/8813134/6772920/dda6e1d6-d0de-11e4-95a8-aeaa63dbb800.PNG)

I checked the code in index.html and find that you commented out the link code around line 45. However, resuming this part of code still don't make the link work. Can you help me with this? Thanks!

**Comment by tombaker1** [comment]
Hi Yao,

The browser interface only works if the UI is loaded from the same server
as the Eden server.  It works if you use localhost as the url.  The browser
will not allow loading data from a different server as the page that is
loaded.  Does this help?

Tom

**Comment by yaooluu** [comment]
Hi Tom,
Thanks for the reply. I did cloned it into my Sahana Eden project's folder as you suggested  and ran it on the same tomcat server with Eden. And my URL is http://localhost:8000/eden/static/EdenMobile/www/index.html, but the problem still remains as above mentioned. Do you know what's going on here?

And besides, I'm currently interested in contribute to Sahana Eden Mobile projects. I found your posts and repo from Google Groups. Here's my recent post on this: https://groups.google.com/forum/?utm_medium=email&utm_source=footer#!topic/sahana-eden/Vh0TqyvS7Dg

Since the J2ME mobile app seems so obselete (2009), do you think you could use some help on your project? If so, to which part of your mobile app can I contribute?

**Comment by tombaker1** [comment]
Hi Yao,

I was just able to recreate your problem.  I have a workaround.  Do the
following:

   - Change line 30 of the config.js file in the root directory to point to
   your local url:

        url: "http://localhost:8000/eden",



   - You also will have to delete the url that is stored locally by your
   browser.  The app uses localstorage. If you are using the Chrome browser
   then open up developer tools for that page 'More tools'->'Developer
   tools'.  Select the Resources tab.  On the left side select LocalStorage,
   and http://localhost:8000 (or whatever your local url is).  Find the
   'user-info' entry and delete that.
   - Reload the page.

You have uncovered several problems.  The first one is that the app doesn't
handle the load error correctly.  Then the app doesn't allow user input
after a load request has been made.  The app can take a long time to load
everything from a slow server.  I see that with the Android app that is
communicating with the Sahana demo server.  The 'loading forms' message is
up for a long time, and the user can't do anything during that time.  This
should be done in the background.  Finally when the app first starts it
stores the 'user-info' into the localstorage, and there is no way to reset
it unless you can get to the settings page.  Thanks for finding these
problems.  :-)  I'll plan to fix them.

In the future we should communicate in the Sahana google group so that
everybody will benefit from our knowledge sharing.

Tom

**Comment by tombaker1** [comment]
> And besides, I'm currently interested in contribute to Sahana Eden Mobile
>> projects. I found your posts and repo from Google Groups. Here's my recent
>> post on this:
>> https://groups.google.com/forum/?utm_medium=email&utm_source=footer#!topic/sahana-eden/Vh0TqyvS7Dg
>>
>> Since the J2ME mobile app seems so obselete (2009), do you think you
>> could use some help on your project? If so, to which part of your mobile
>> app can I contribute?
>>
>
Yes, I could use a lot of help.  I am in the process of documenting the
design of the app so that others can help.  I am currently working on the
features for the shelters page.  The design is modular so new features, and
pages can be worked on independently.  So you can help add new features to
the shelters page, or copy the design for the shelters page to create a new
one such as warehouses, hospitals, etc.

Tom

**Comment by yaooluu** [comment]
Thanks for the reply! The workaround works, and I can run your project now. I got your direction and am working on this now :-)

**Comment by yaooluu** [comment]
Hello Tom!

Below is a glimpse of the form generator page I created. You can choose language, then form name, and the attributes you want, then after clicking the generate button, the textarea on the right will generate the table variable (in JSON format). I will clean up the code and send you a pull request soon.
![1](https://cloud.githubusercontent.com/assets/8813134/7478334/1b266dae-f328-11e4-9517-b6b2b2959986.png)

By the way, could you please help giving some comments on my team's work so far? Actually I'm a student and this is part my course project. Your comments will be considered for evaluating our team's work. It would be so kind if you could just write some short comments and send it to our course TA. Here's his my TA Govind's email: grmenon@ncsu.edu. Thank you very much and I really appreciate your effort!

Best,
Yao

**Comment by yaooluu** [comment]
Hi Tom, 

I just finished the demo of our team's work in front my professor. He took your comment very helpful for evaluating our work. Actually, he has a link containing a simple evaluation form that needs to be filled by someone who is in charge of this project. Could you please help filling this part? And the comment paragraph won't necessary. And can you give me an email address for sending the form link to? My email address is ylu31@ncsu.edu. Thank you very much!

Yao.
